src/cljs/athens/events.cljs

```diff
@@ -39,6 +39,133 @@
+                                        :block/order    athens-child-count
+                                        :block/open     true})
+        merge-pages              (merge page-roam page-athens)
+        final-page-with-children (assoc merge-pages :block/children new-children)]
+    (if (zero? roam-child-count)
+      merge-pages
+      final-page-with-children)))
+
+
+(defn get-shared-pages
+  [roam-db]
+  (->> (d/q '[:find [?pages ...]
+              :in $athens $roam
+              :where
+              [$athens _ :node/title ?pages]
+              [$roam _ :node/title ?pages]]
+            @athens.db/dsdb
+            roam-db)
+       sort))
+
+
+(defn pages
+  [roam-db]
+  (->> (d/q '[:find [?pages ...]
+              :in $
+              :where
+              [_ :node/title ?pages]]
+            roam-db)
+       sort))
+
+
+(defn gett
+  [s x]
+  (not ((set s) x)))
+
+
+(defn not-shared-pages
+  [roam-db shared-pages]
+  (->> (d/q '[:find [?pages ...]
+              :in $ ?fn ?shared
+              :where
+              [_ :node/title ?pages]
+              [(?fn ?shared ?pages)]]
+            roam-db
+            athens.events/gett
+            shared-pages)
+       sort))
+
+
+(defn update-roam-db-dates
+  "Strips the ordinal suffixes of Roam dates from block strings and dates.
+  e.g. January 18th, 2021 -> January 18, 2021"
+  [db]
+  (let [date-pages         (d/q '[:find ?t ?u
+                                  :keys node/title block/uid
+                                  :in $ ?date
+                                  :where
+                                  [?e :node/title ?t]
+                                  [(?date ?t)]
+                                  [?e :block/uid ?u]]
+                                db
+                                athens.patterns/date-block-string)
+        date-block-strings (d/q '[:find ?s ?u
+                                  :keys block/string block/uid
+                                  :in $ ?date
+                                  :where
+                                  [?e :block/string ?s]
+                                  [(?date ?s)]
+                                  [?e :block/uid ?u]]
+                                db
+                                athens.patterns/date-block-string)
+        date-concat        (concat date-pages date-block-strings)
+        tx-data            (map (fn [{:keys [block/string node/title block/uid]}]
+                                  (cond-> {:db/id [:block/uid uid]}
+                                          string (assoc :block/string (athens.patterns/replace-roam-date string))
+                                          title (assoc :node/title (athens.patterns/replace-roam-date title))))
+                                date-concat)]
+    ;;tx-data))
+    (d/db-with db tx-data)))
+
+;;(/ 3736 3842) 97% clean
+;;(-> (- 1056 2)
+;;    (+ (- 3088 406))))
+;;(defonce ROAM-DB (atom nil))
```

Comment:
🤔   This line of comment might not be intentional?

![Screen Shot 2021-01-29 at 5 54 46 PM](https://user-images.githubusercontent.com/5305984/106303998-16437e00-625b-11eb-8741-28e2abd20d3c.png)

Reply:
Yes not intentional

Reply:
Great, the roam import worked for me after uncommenting the line.

Reply:
How's performance? What is your `index.transit` size now? @avichalp

Reply:
```yaml
index.transit: 4.1M
roam export EDN file`: 7.2M
```

There is a high latency in loading search results and in loading pages with a high number of linked blocks (~ 400+). From my perception, latency is as high as 10-11 seconds sometimes. Would like to know how is the performance for other people. If we were to optimize performance, the first step would be to collect some data, I guess.

Reply:
My guess is that this stems from inefficient posh queries. Roam and LogSeq teams both told me that Posh wasn't super performant, so they both ended up writing their own reactive lightweight wrappers around Datascript.

I also remember that @jeroenvandijk was working on performance optimizations a while ago, using Reagent cursors. I wonder if that is part of the solution. https://github.com/athensresearch/athens/pull/93/files#diff-3c7f15f69987f2ac41d3dfa65d60e8dfae0778e494b2207f64721d20baad680cR43-R46


Current performance issue: #570